llama_index/tools/function_tool.py

```diff
@@ -0,0 +1,78 @@
+from typing import Any, Optional, cast, Callable
+
+from llama_index.indices.query.base import BaseQueryEngine
+from llama_index.langchain_helpers.agents.tools import IndexToolConfig, LlamaIndexTool
+from llama_index.tools.types import BaseTool, ToolMetadata
+from langchain.tools import Tool, StructuredTool
+
+DEFAULT_NAME = "Function Tool"
+DEFAULT_DESCRIPTION = """Useful for running a natural language query
```

Comment:
Does it make sense to have a default name and description? Would a user ever leave this as the default?

Reply:
oh yeah i guess the defautl should really just be the function signature...similar to what we do in toolspec